### PR TITLE
Allow different duration types to specified within config files.

### PR DESCRIPTION
**Pull request by pocockn**
### Problem
Durations were being set as nanoseconds and then converted into seconds which would make the config confusing. 

### Solution
Allow the duration type to be selected in the config with the use of struct tags. 

For example:
`TestTimeHour  time.Duration `config:"test_time_hours" config_duration_type:"hours"`

### Notes

- **Remember** to tag your merge commit after merging 👍
- **Remember** to add the Clubhouse ticket url as a comment ♣️ 🏠

### CODE REVIEW

**Comment by pocockn** [comment]
https://app.clubhouse.io/vidsy/story/5994/new-briefdeadlineeventpublisher-service

**Comment by pocockn** [comment]
https://app.clubhouse.io/vidsy/story/6062/add-time-duration-as-a-type-which-can-be-used-in-our-service-configs

File: kmsconfig/config_test.go [hunk 1]
```diff
@@ -117,7 +117,9 @@ func TestConfig(t *testing.T) {
 					TestString      string        `config:"test_string"`
 					TestStringSlice []string      `config:"test_string_slice"`
 					TestInt         int64         `config:"test_int"`
-					TestTime        time.Duration `config:"test_time"`
+					TestTimeSecond  time.Duration `config:"test_time" config_duration_type:"seconds"`
```
Comment (revett) [inline kmsconfig/config_test.go]: Can the struct tags be formatted like this? https://github.com/vidsy/go-models/blob/master/api/data/brief.go#L14

File: kmsconfig/config_test.go [hunk 2]
```diff
@@ -113,11 +113,16 @@ func TestConfig(t *testing.T) {
...
-					TestBool        bool          `config:"test_bool"`
-					TestString      string        `config:"test_string"`
-					TestStringSlice []string      `config:"test_string_slice"`
-					TestInt         int64         `config:"test_int"`
-					TestTime        time.Duration `config:"test_time"`
+					TestBool            bool          `config:"test_bool"`
+					TestString          string        `config:"test_string"`
+					TestStringSlice     []string      `config:"test_string_slice"`
+					TestInt             int64         `config:"test_int"`
+					TestTimeMicrosecond time.Duration `config:"test_time_microseconds"      config_duration_type:"microseconds"`
```
Comment (revett) [inline kmsconfig/config_test.go]: These fields don't need as much space between the struct tags:

```go
TestTimeMicrosecond time.Duration `config:"test_time_microseconds" config_duration_type:"microseconds"`
TestTimeMillisecond time.Duration `config:"test_time_milliseconds" config_duration_type:"milliseconds"`
TestTimeSecond      time.Duration `config:"test_time_seconds"      config_duration_type:"seconds"`
TestTimeMinute      time.Duration `config:"test_time_minutes"      config_duration_type:"minutes"`
TestTimeHour        time.Duration `config:"test_time_hours"        config_duration_type:"hours"`
TestTimeDay         time.Duration `config:"test_time_days"         config_duration_type:"days"`
```

File: kmsconfig/config_test.go [hunk 3]
```diff
@@ -128,7 +133,34 @@ func TestConfig(t *testing.T) {
...
+			assert.Equal(t, time.Duration(2*time.Second), configStruct.App.TestTimeSecond)
+			assert.Equal(t, time.Duration(5*time.Minute), configStruct.App.TestTimeMinute)
+			assert.Equal(t, time.Duration(10*time.Hour), configStruct.App.TestTimeHour)
+			assert.Equal(t, time.Duration((time.Hour*24)*11), configStruct.App.TestTimeDay)
+		})
+
+		t.Run("ReturnsErrorIfConfigDurationTypeTagIsInvalid", func(t *testing.T) {
+			var configStruct struct {
+				App struct {
+					TestTimeDay time.Duration `config:"test_time_days"  config_duration_type:"error"`
```
Comment (revett) [inline kmsconfig/config_test.go]: Here also only a single space is needed between the struct tags:

```
`config:"test_time_days" config_duration_type:"error"`
```

File: kmsconfig/config_test.go [hunk 4]
```diff
@@ -113,11 +113,16 @@ func TestConfig(t *testing.T) {
...
-					TestInt         int64         `config:"test_int"`
-					TestTime        time.Duration `config:"test_time"`
+					TestBool            bool          `config:"test_bool"`
+					TestString          string        `config:"test_string"`
+					TestStringSlice     []string      `config:"test_string_slice"`
+					TestInt             int64         `config:"test_int"`
+					TestTimeMicrosecond time.Duration `config:"test_time_microseconds"      config_duration_type:"microseconds"`
+					TestTimeMillisecond time.Duration `config:"test_time_milliseconds"      config_duration_type:"milliseconds"`
+					TestTimeSecond      time.Duration `config:"test_time_seconds"           config_duration_type:"seconds"`
+					TestTimeMinute      time.Duration `config:"test_time_minutes"           config_duration_type:"minutes"`
```
Comment (revett) [inline kmsconfig/config_test.go]: The struct field name and the JSON config key node should be the same.

```
TestTimeMinute vs test_time_minutes
```

**Pull request merged**